### ISSUE TITLE
results.Opt as object fields could generate codegen issues

### ISSUE THREAD

**Issue by cheatfate** [status-im/nim-stew#142]
All Nim versions are affected
https://github.com/status-im/nim-stew/pull/141

```nim
import stew/results

type
  ValidatorSig = object
    blob: array[96, byte]
  SyncDutyAndProof = object
    slotSig: Opt[ValidatorSig]
const
  DefaultSyncDutyAndProof =
    SyncDutyAndProof(slotSig: Opt.none(ValidatorSig))

echo DefaultSyncDutyAndProof
```

```
/home/runner/.cache/nim/all_tests_d/@mtest_results.nim.c:974:143: error: ‘union <anonymous>’ has no member named ‘_o_1’; did you mean ‘_o_2’?
 N_LIB_PRIVATE NIM_CONST tyObject_SyncDutyAndProof__9atwCZvu9aDnhulGNwFXJ9cFg DefaultSyncDutyAndProof__qmlenLhfD5mHF063hF585Q = {{NIM_FALSE, {._o_1 = {}}}
                                                                                                                                               ^~~~
                                                                                                                                               _o_2
                                                                                                                                               
```

**Comment by deech** [comment]
Seems like a bug in [Nim itself](https://github.com/nim-lang/Nim/issues/20699)

**Comment by deech** [comment]
https://github.com/nim-lang/Nim/pull/20790 should fix it.

**Comment by arnetheduck** [comment]
This is fixed both by #150 and upstream fixes
